### ISSUE TITLE
Only one database can be connected/attache at the same time to browser

### ISSUE THREAD

**Issue by lhurt** [keepassxreboot/keepassxc-browser#659]
## Expected Behavior
Multiple databases can be connected/attached to the browser simultaneously for parallel usage.

## Current Behavior
Existing connections get overwritten, practically preventing the simultaneous usage of multiple databases.

## Steps to Reproduce (for bugs)
Assumption is that no database is connected at all.
`Search in all opened databases for matching credentials` is checked
1. Open database A and B in KeepassXC, tab of database b is active
2. Open Settings page of browser add-on and go to "connected databases"
3. Click "connect" and add key name "Key B" in pop up window
4. The new connection appears in the list
5. Change tab to database A in KeepassXC
6. Open Settings page of browser add-on and go to "connected databases"
7. Click "connect" and add key name "Key A" in pop up window
8. The new connection appears in the list
9. Close Settings window
10. Open Settings page of browser add-on and go to "connected databases"
11. Only the last database connection is listed and used by the add-on

## Debug info

KeePassXC - 2.5.0
KeePassXC-Browser - 1.5.3
Operating system: Win/Linux
Browser: Chrome/Firefox/
Tested any combination with the same results

Please see also issues #318 #634 [#3485 in keepassxc repo](https://github.com/keepassxreboot/keepassxc/issues/3485)
They are closed but the described issue still exists.

**Comment by varjolintu** [comment]
I made exactly those steps using `web-ext` so it's always a new profile for Firefox. The Connected Databases tab always shows both databases connected, plus I tested a GitHub login page and it searches credentials from both databases.

Can you test this with a new temporary profile? This bug sounds a bit strange because 1.5.3 was the version that fixes these kind of problems.

**Comment by lhurt** [comment]
Unfortunately creating new profiles doesn't do the job. Tested it on Linux in Chrome and Firefox.

1. Created a new profile
2. Added only the keepassxc extension
3. Did the steps as described above
4. Behavior the same as described above

I'd be more than willing to help if possible.

BTW. My settings in KeePassXC in the browser integration panel are
- General
  - Enabled Google Chrome, Firefox, Chromium
  - [ON] Request to unlock the database if it is locked
  - [OFF] Return only best matching credentials
  - [OFF] Allow returning expired credentials
  - Sort matching credentials by title
- Advanced
  - [ON] Never ask before accessing credentials
  - [OFF] Never ask before updating credentials
  - [ON] Search in all opened databases for matching credentials
  - [ON] Return advanced string fields start with "KPH:"
  - [OFF] Do not prompt for KeepassHTTP settings migration
  - [OFF] Update native messaging manifest files at startup
  - [ON] Use a proxy application between KeePassXC and browser extension
  - [OFF] Use a custom proxy location

Thanks very much

**Comment by lhurt** [comment]
Yep. You nailed it. Actually I had a very big database and split it up in three by copying and deleting the unwanted entries.
Now I created new databases and moved the items and it works now perfectly.

Checked it on Linux and Windows with Firefox and Chrome.

So sorry to cause such a hassle, but finally you cut it.

Thanks so much for your patience and your help.

**Comment by gotjoshua** [comment]
@varjolintu 
> probably have to create a possibility to update the root group UUID
any progress on that?

It took a fair bit of time and googling to realize that this wasn't working due to common root UUID via db duplication
: (
